packages/x-date-pickers-pro/src/DateTimeRangePicker/shared.tsx

```diff
@@ -122,6 +122,11 @@ export interface BaseDateTimeRangePickerProps<TDate extends PickerValidDate>
    * If `undefined`, internally defined view will be used.
    */
   viewRenderers?: Partial<DateTimeRangePickerRenderers<TDate, DateTimeRangePickerView>>;
+  /**
+   * If `true`, the popover or modal will close after submitting the full date.
+   * @default false
+   */
+  closeOnSelect?: boolean;
```

Comment:
Maybe we could update the JSDoc of this prop on `UsePickerValueNonStaticProps` type definition since it is now the most common default.
Then we can keep the overrides only for `DateRangePicker` and `DesktopDateRangePicker`. 🤔 
WDYT?
```suggestion
```